src/Prometheus.Client.HttpRequestDurations/HttpRequestDurationsMiddleware.cs

```diff
@@ -46,7 +46,15 @@ public HttpRequestDurationsMiddleware(RequestDelegate next, HttpRequestDurations
 
         public async Task Invoke(HttpContext context)
         {
-            var path = NormalizePath.Execute(context.Request.Path, _options);
+            var path = string.Empty;
+
+#if HasRoutes
+            if (_options.UseRouteName)
+                path = context.GetRouteName();
```

Comment:
Why GetRouteName without NormalizePath?

Reply:
Do we need to normalize it? As route name or route pattern is being taken from source code of the application.

Reply:
Custom normalization.

Like this:
```
    q.CustomNormalizePath = new Dictionary<Regex, string>
    {
        { new Regex(@"\/[0-9]{1,}(?![a-z])"), "/id" } // Replace 'int' in Route
    };
```

Reply:
Done